ampe.c

```diff
@@ -816,7 +817,9 @@ static void fsm_step(struct candidate *cand, enum plink_event event)
                     cand->mtk, sizeof(cand->mtk),
                     cand->mgtk, sizeof(cand->mgtk),
                     cand->mgtk_expiration,
-                    cand->igtk, sizeof(cand->igtk), cand->igtk_keyid,
+                    (cand->has_igtk) ? cand->igtk : NULL,
+                    (cand->has_igtk) ? sizeof(cand->igtk) : 0,
```

Comment:
It will actually work in this case because igtk is a char[16] in struct candidate.  But I take your point that it is a little fragile, a #define for array decl and here would be better.

Reply:
We're using sizeof() in a bunch of other codepaths, so I'll add the #define in the array declaration and a couple of open-coded 16s, but will keep the sizeof() here.